dist/src/main/java/io/camunda/zeebe/shared/management/BackupEndpoint.java

```diff
@@ -65,11 +70,7 @@ public WebEndpointResponse<?> take(final long backupId) {
                       .formatted(backupId, backupId)),
           202);
     } catch (final CompletionException e) {
-      final var errorCode =
-          e.getCause() instanceof BackupAlreadyExistException
-              ? 409
-              : WebEndpointResponse.STATUS_INTERNAL_SERVER_ERROR;
-      return new WebEndpointResponse<>(new Error().message(e.getCause().getMessage()), errorCode);
+      return mapErrorResponse(e.getCause());
     } catch (final Exception e) {
       return new WebEndpointResponse<>(
```

Comment:
:wrench: How about merging this into `mapErrorResponse`? So that `mapErrorResponse` takes any exception, and unwraps `CompletionException` if necessary.

This would also reduce the number of tests because `shouldReturnErrorOnCompletionException` and `shouldReturnErrorOnException` wouldn't be needed anymore because normal exceptions could be part of the `Failures` argument provider.

Reply:
Merged the error handling. But cannot remove test `shouldReturnErrorOnException`. We need different ways to mock the api for this test and for `shouldReturnErrorOnCompletionException`. But I merged `shouldReturnErrorOnCompletionException` it with the parameterized test.